### ISSUE TITLE
Scene text detection and recognition in “the wild” (natural scenes)

### ISSUE THREAD

**Issue by raamb** [singnet/rfai-proposal#9]
**Author**
Oleg Shcherbakov

**Description**
- Service that reads (detects and recognises) the text in the input image(s). 
- Service has to be able to process input image and find all the text in it and output the text in suitable form.

Task of scene text detection and recognition refers to the research area dealing with the interpretation of written or printed communication in unconstrained settings. This research field is also called “Robust Reading”. This task lies between document imaging analysis and scene interpretation in the wide sense. This task serves as a good proving ground for the complex computer vision systems. 
Since 2003 there are Robust Reading Competitions (https://rrc.cvc.uab.es/) which are aiming to draw attention to this field of research. Such competitions involve different subtasks on the variety of datasets. Also there are few extremely challenging problems like end-to-end test detection and recognition and end-to-end text spotting. 
Typical pipeline for this task is built upon deep CNN architectures (http://www.robots.ox.ac.uk/~vgg/publications/2014/Jaderberg14/jaderberg14.pdf , https://arxiv.org/pdf/1801.01315.pdf and etc.)
However, each approach uses its own heuristics for improving one of the two critical parts of the pipeline: either detection or recognition. Such heuristics commonly consist of some post processing tricks like adaptive non-maximum suppression (https://arxiv.org/pdf/1704.03155v2.pdf), adoption of classic computer vision algos like contour linkage and segment linkage (https://arxiv.org/pdf/1703.06520.pdf) or accounting for the space geometry inside the trainable models (https://arxiv.org/pdf/1712.05404.pdf). 
Typical CNN architectures are based on the popular object detection models, such as YOLO, R-CNN or SSD.

Requested service should be able to process an input image to output all the text data contained in the image. The output text data should be grouped in the relevant contextual frames, e. g. for this image there may be 3 frames consisting of text on the plaques or may be 9 frames corresponding to the number of text lines on the image. 

**Refrences**
Baseline models
- eccv2014_textspotting
https://bitbucket.org/jaderberg/eccv2014_textspotting/src/master/
Authors make the following contributions: they develop a Convolutional Neural Network (CNN) classifier that can be used for detecting words regions in the image, and recognizing the words within these regions. “The CNN has a novel architecture that enables efficient feature sharing (by using a number of layers in common) for text detection, character case-sensitive and insensitive classification, and bigram classification.”
- PixelLink: Detecting Scene Text via Instance Segmentation
https://github.com/ZJULearning/pixel_link
A novel scene text detection algorithm based on instance segmentation, is proposed.
- SEE: Towards Semi-Supervised End-to-End Scene Text Recognition
https://github.com/Bartzi/see
Authors propose to use a single deep neural network, that learns to detect and recognize text from natural images, in a semi-supervised way.
- EAST: An Efficient and Accurate Scene Text Detector
https://github.com/argman/EAST
“The pipeline directly predicts words or text lines of arbitrary orientations and quadrilateral shapes in full images, eliminating unnecessary intermediate steps, with a single neural network.”
- SegLink
https://github.com/bgshih/seglink
The main idea of the method “is to decompose text into two locally detectable elements, namely segments and links. A segment is an oriented box covering a part of a word or text line; A link connects two adjacent segments, indicating that they belong to the same word or text line. Both elements are detected densely at multiple scales by an end-to-end trained, fully-convolutional neural network.”
- TextBoxes: A Fast Text Detector with a Single Deep Neural Network
https://github.com/MhLiao/TextBoxes
This method “presents an end-to-end trainable fast scene text detector, named TextBoxes, which detects scene text with both high accuracy and efficiency in a single network forward pass, involving no post-process except for a standard non-maximum suppression.”
- TextProposals
https://github.com/lluisgomez/TextProposals
Implementation of the method "TextProposals: a Text-specific Selective Search Algorithm for Word Spotting in the Wild" (Gomez and Karatzas)
- DeepTextSpotter
https://github.com/MichalBusta/DeepTextSpotter
The structure of the recognition CNN and the geometry of its input layer preserves the aspect of the text and adapts its resolution to the data.
- Word Spotting and Recognition with Embedded Attributes
https://github.com/almazan/watts
Authors “propose to use character attributes to learn a semantic representation of the word images and then perform a calibration of the scores with CCA that puts images and text strings in a common subspace.”

Papers

- End-to-End Interpretation of the French Street Name SignsDataset
https://arxiv.org/pdf/1702.03970.pdf
“Dataset consisting of more than a million images of street name signs cropped from Google Street View images of France. Each image contains several views of the same street name sign.”

**Training Datasets**
There is plenty of data related to this task. We cite the most relevant ones:
- COCO-Text is a new large scale dataset for text detection and recognition in natural images.
 (https://vision.cornell.edu/se3/coco-text-2/)
- TOTAL-TEXT is a word-level based English curve text dataset. If you are interested in text-line based dataset with both English and Chinese instances, we highly recommend you to refer SCUT-CTW1500. 
(https://github.com/cs-chan/Total-Text-Dataset)
- ICDAR 2019 Robust Reading Challenge on Multi-lingual scene text detection and recognition
This is a very challenging task of a unified OCR for multiple-languages. The end-to-end scene text detection and recognition task in multi-language setting is coherent with its English counterparts. Given an input scene image, the objective is to localize a set of bounding boxes and their corresponding transcriptions.
(https://rrc.cvc.uab.es/?ch=15&com=evaluation&task=4) (See also 
ICDAR2019 Robust Reading Challenge on Arbitrary-Shaped Text
ICDAR2019 Robust Reading Challenge on Large-scale Street View Text with Partial Labeling
ICDAR 2019 Robust Reading Challenge on Scanned Receipts OCR and Information Extraction
ICDAR 2019 Robust Reading Challenge on Reading Chinese Text on Signboard)
- NEOCR: Natural Environment OCR Dataset contains 659 real world images with 5238 annotated bounding boxes (textfields). The images were taken by several people independently from the dataset, so the dataset covers a broad range of characteristics which distinguish real world images from scanned documents. All text recognizable by humans has been annotated for all images. The dataset creation process was stopped when for each metadata dimension at least 100 textfields were included in the dataset.
(http://www.iapr-tc11.org/mediawiki/index.php?title=NEOCR:_Natural_Environment_OCR_Dataset)
- KAIST Scene Text Database comprises 3000 images captured in different environments, including outdoors and indoors scenes under different lighting conditions (clear day, night, strong artificial lights, etc). Images were captured either by the use of a high-resolution digital camera or a low-resolution mobile phone camera. 
(http://www.iapr-tc11.org/mediawiki/index.php?title=KAIST_Scene_Text_Database)
- Brno Mobile OCR Dataset (B-MOD) is a collection of 2 113 templates (pages of scientific papers). Those templates were captured using 23 various mobile devices under unrestricted conditions ensuring that the obtained photographs contain various amount of blurriness, illumination etc. In total, the dataset contains 19 725 photographs and more than 500k text lines with precise transcriptions. The template pages are divided into three subsets (training, validation and testing).
(https://pero.fit.vutbr.cz/brno_mobile_ocr_dataset)

**Acceptance Criteria**
- The output text data should be grouped in the relevant contextual frames. The ​[COCO-Text](https://vision.cornell.edu/se3/coco-text-2/) annotation format​ can be considered as an example.
- Metrics: Typical metrics are recall, precision and f-score. Currently available competitive solutions achieve less than 75% precision and 55% recall with average precision less than 45% on the challenging task of end-to-end test detection and recognition (https://rrc.cvc.uab.es/?ch=15&com=evaluation&task=4). Those metrics can be considered as the nearest milestone for developers of such systems.
- Minimum Target Metric:  Accomplishment of 80% of F1-score on both the Task-4 of RRC-MLT-2019 and Task-3 for COCO-text dataset. (Achieves at least 80% precision and 60% recall on the Task-4 of RRC-MLT-2019 and at least 75% average precision on the Task-3 for COCO-text dataset)

**Reward**
20000 tokens

**Expiration Date**
10 May 2020

**Non-functional Requirements**
All codes of the proposed solution have to be OSS.
